src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -49,6 +50,40 @@ public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Tra
                         .SetMinimumLevel(RazorLSPOptions.GetLogLevelForTrace(trace)))
                     .OnInitialized(async (s, request, response) =>
                     {
+                        bool objectSupportedVersion;
+
+                        if (Version.TryParse(request.ClientInfo?.Version, out var parsedVersion))
+                        {
+                            version = parsedVersion;
+                        }
+
+                        if (version is null)
+                        {
+                            version = new Version("1337");
```

Comment:
I'd move this version into a `private static readonly Version MaxSupportedProtocol = new Version(1337, 0, 0, 0)` with a comment about us picking a large number to emulate "latest" support and do

```suggestion
                            version = MaxSupportedProtocol;
```